internal/chain/types.go

```diff
@@ -66,13 +66,22 @@ type Transaction struct {
 const emptyInput = "0x"
 const emptyReceiver = ""
 
-func GetTransactionType(to, input string, internal bool) models.TransactionType {
+func GetInternalTransactionTarget(trace TraceEntry) string {
+	if trace.Action.To != "" {
+		return trace.Action.To
+	}
+	return trace.Result.Address
+}
+
+func GetTransactionType(to, input, txType string, internal bool) models.TransactionType {
 	trx_type := 0
 	// add offset if transaction is internal
 	if internal {
 		trx_type = 3
+		if txType == "create" {
+			return models.InternalContractCreation
+		}
```

Comment:
So if ifs from below isn't working on internal transactions lets have totally separate logic for it here for calls and creates also

Reply:
```suggestion
if internal {
		trx_type = 3
		if txType == "create" {
			return models.InternalContractCreation
		} else if txType = "call" {
		    ....
		}
		return models.InternalContractCall
} else {
	if to == emptyReceiver {
		return models.ContractCreation
	} else if input != emptyInput {
		return models.ContractCall
	}
	return models.Transfer
}
```